GitCommands/Patches/PatchProcessor.cs

```diff
@@ -30,13 +30,13 @@ private enum PatchProcessorState
         // diff --git a/GitCommands/CommitInformationTest.cs b/GitCommands/CommitInformationTest.cs
         // diff --git b/Benchmarks/App.config a/Benchmarks/App.config
         // diff --cc config-enumerator
-        [GeneratedRegex(@$"^diff --(?<type>git|cc|combined)\s[""]?([abiwco12]/)?(?<filenamea>.*?)[""]?( [""]?[abiwco12]/(?<filenameb>.*?)[""]?)?\s*$", RegexOptions.ExplicitCapture)]
```

Comment:
PatchProcessor need this, as patches could be created by other tools than GE
For submodule helpers and in general, the Git setting should be overridden. The line of changes is getting long though....